### ISSUE TITLE
Stock recruitment Fig 31 

### ISSUE THREAD

**Issue by andrew-edwards** [pacific-hake/hake-assessment#917]
1. Should `R_0 (billions)` y-axis in right panel not have units, since it's relative to R_0? If so I think just leave the label off (or duplicate the main y-axis label).
2. And ideally the scale would be the same as the main figure (which may indeed be too fiddly), yes?  
3. And anyone tried fitting a regression to see what the slope is? ;)

**Comment by andrew-edwards** [comment]
Looking into it, I now think that the right panel is absolute recruitment, which is why it shows 2.5 level with 1 on the 'relative to R0' axis of the main plot. Last year it was 2.3 (and lower number was 0.8, it happens to be 1 this year).
So think that 
- [ ]'R0 (billions)' should be 'Recruitment (billions)'
- [ ] vertical text should be 'Distribution of absolute recruitments'
- [ ] end of caption needs changing also.
Can't see the /R0 in the plotting function early on (it plots the right panel first). So the numbers are correct I think, just mislabelled.
I can do tomorrow if anyone agrees with my thinking.....

**Comment by cgrandin** [comment]
I checked the code as well and the right panel is the median of the `Recr_Virgin` posterior divided by 1 million so that is correct although maybe it should not be billions since it's divided by 1e6: https://github.com/pacific-hake/hake-assessment/blob/9bc5898829f3a5a2e0c3a02cc0ee8a562cffed5f/R/figures-stock-recruitment.R#L95

The graph on the right is the normal distribution with mean = `(-sigma_r ^2) / 2` and SD = `sigma_r` and the caption says "Expected distribution" so I think the text is correct (we could add "Expected" to the plot label)

**Comment by andrew-edwards** [comment]
Units of recr are saved as 1000s (I had to check this the other day). So 2.5 billion fish gets saved as 2.5 million in units of 000s of fish, so dividing by million gives 2.5 which is correctly in billions of fish. 🤯  
- [ ] Yes, 'Expected' is okay. I'm thinking "expected distribution of absolute recruitment". Not relative.

**Comment by andrew-edwards** [comment]
See above commits, shows what I meant. Feel free to close.

**Comment by andrew-edwards** [comment]
Pretty sure this is right now. Chris thought Allan made the figure so see if he says anything.

**Comment by andrew-edwards** [comment]
Hmmmmm..............

**Comment by andrew-edwards** [comment]
Consensus seems to be to scrap the right-hand figure. This justifies it (more than 'no-one understands it'):
```
> R_2022 <- base.model$mcmc[["Recr_2022"]]/1e6 # billions of fish
> quantile(R_2022, c(0, 0.5, 0.9, 0.95))
         0%         50%         90%         95% 
 0.00445245  0.92389500  7.26290600 12.42756000 
> hist(R_2022, breaks = 500, xlim = c(0,10))
> hist(R_2022, breaks = 500)
```
 
![image](https://user-images.githubusercontent.com/6977190/154371790-c021c5d8-069d-4ecc-ba8a-c549c7837c5e.png)

So, note the x-axis. Top one 'only' goes up to 10 billion and has some support. But bottom one is the full range, out to the top values (for which can't the histogram bars).

Then:
```
> hist(R_2022, breaks = 100)
> hist(R_2022, breaks = 100, ylim = c(0,1000))
> hist(R_2022[R_2022 > quantile(R_2022, 0.9)], breaks = 100, add = TRUE, col = "red")
> hist(R_2022[R_2022 > quantile(R_2022, 0.95)], breaks = 100, add = TRUE, col = "blue")
```
![image](https://user-images.githubusercontent.com/6977190/154372249-28d55906-f7d8-4c44-ae35-7e8f1cbbaed8.png)

**Comment by aaronmberger-nwfsc** [comment]
SRG said not to scrap it, but fix it (or look further into it).

**Comment by andrew-edwards** [comment]
Well, having looked into it further can we conclude to scrap it? I think the problem is that however it's plotted it's just misleading to the eye, because the tail is so long but thin....

**Comment by andrew-edwards** [comment]
So I think it's just not useful, and misleading.

**Comment by andrew-edwards** [comment]
What are we thinking? Leave as is and work on Aaron's idea next year? Unless someone wants to go at it today (I don't!).

**Comment by aaronmberger-nwfsc** [comment]
As I see the SRG has a similar request as last year in their draft report related to sampling recruitment from observations or a mixture distribution, it tells me that we can address this figure along with that next year.  So perhaps we milestone this one until then.

**Comment by aaronmberger-nwfsc** [comment]
Okay, I think I found the bug in this plot code.  When calculating the bias adjustment factor it has -sigma^2/2, but it supposed to be sigma^2/2.  I did a test and the results are below.  Top figure is our usual and the bottom figure is the proposed fix.  The distribution for the right hand panel plot looks more like we all were expecting.  I'm going to fix and commit this, but will leave open for now for discussion.
![test_old](https://user-images.githubusercontent.com/7946234/214485679-9c523ac3-3fd8-46e9-9bb2-0213480e1c53.png)
![test_new](https://user-images.githubusercontent.com/7946234/214485693-a094a475-f4e4-423a-be82-55f75e6ad9af.png)
